ui/page/transaction/transactions_page.go

```diff
@@ -222,93 +219,135 @@ func (pg *TransactionsPage) loadTransactions(offset, pageSize int32) ([]*sharedW
 // Layout draws the page UI components into the provided layout context
 // to be eventually drawn on screen.
 // Part of the load.Page interface.
-func (pg *TransactionsPage) Layout(gtx layout.Context) layout.Dimensions {
+func (pg *TransactionsPage) Layout(gtx C) D {
 	if pg.Load.GetCurrentAppWidth() <= gtx.Dp(values.StartMobileView) {
 		return pg.layoutMobile(gtx)
 	}
 	return pg.layoutDesktop(gtx)
 }
 
-func (pg *TransactionsPage) layoutDesktop(gtx layout.Context) layout.Dimensions {
+func (pg *TransactionsPage) walletNotReady() bool {
+	return !pg.selectedWallet.IsSynced() || pg.selectedWallet.IsRescanning()
+}
+
+func (pg *TransactionsPage) showDisabledLayout(gtx C, body layout.Widget) D {
+	overlay := layout.Stacked(func(gtx C) D {
+		if pg.walletNotReady() && pg.isHomepageLayout {
+			gtx = gtx.Disabled()
+			return components.DisablePageWithOverlay(pg.Load, nil, gtx, values.String(values.StrFunctionUnavailable), nil)
+		}
+		return D{}
+	})
+
+	return layout.Stack{}.Layout(gtx, layout.Expanded(body), overlay)
+}
```

Comment:
3 problems I see here.

1. This `showDisabledLayout` method doesn't always show a disabled layout. Maybe rename the method or remove it entirely.
2. If there's no need to disable the layout, no need to do `layout.Stack{}.Layout(gtx, layout.Expanded(body), overlay)` with an empty overlay. Just return the original body!
3. `components.DisablePageWithOverlay` is being abused and this PR should not follow that pattern of abuse. The function is meant to disable a page or a widget and apply an overlay over the widget. If you're not passing a page/widget, then you should not be using that function. Apparently passing a `nil` causes the fn to return the overlay for you to use as you like, let's have another function that gives you an overlay without asking you for a page or widget. It should also be a theme method, so maybe `theme.OverlayWidget(txt, btn) layout.Widget`.